src/components/media-query/withMediaQuery.js

```diff
@@ -1,23 +1,40 @@
 // @flow
 
 import * as React from 'react';
+import classNames from 'classnames';
 import useMediaQuery from './useMediaQuery';
 import type { MediaShape } from './types';
+import { CLASS_IS_MEDIUM, CLASS_IS_SMALL, SIZE_MEDIUM, SIZE_SMALL } from '../../constants';
 
 type PropsShape = {
     children: React.Node,
+    className: string,
 };
 
 function withMediaQuery<Props: PropsShape>(WrappedComponent: React.ComponentType<any>): React.ComponentType<any> {
-    return ({ children, ...rest }: Props) => {
+    const MediaQueryComponent = ({ children, className, ...rest }: Props) => {
         const mediaProps: MediaShape = useMediaQuery();
 
+        const isSmall = mediaProps.size === SIZE_SMALL;
```

Comment:
We generally want to avoid casting enums to booleans, as it creates a situation where we can have multiple conflicting states versus a single known state.